debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/Wal2JsonReplicationMessage.java

```diff
@@ -139,110 +150,213 @@ public String getName() {
      * @return the value; may be null
      */
     public Object getValue(String columnName, String columnType, Value rawValue, final PgConnectionSupplier connection) {
-        switch (columnType) {
+        // we support wal2json both before & after commit 0255f2ac, which means we may get old-style type names
+        // (eg. int2, timetz, varchar, numeric, _int4)
+        // or new-style/verbose ones
+        // (eg. smallint; time with time zone; character varying (255); decimal (10,3), integer[])
+
+        Matcher m = TYPE_PATTERN.matcher(columnType);
+        if (!m.matches()) {
+            LOGGER.error("Failed to parse columnType for {} '{}'", columnName, columnType);
+            throw new ConnectException(String.format("Failed to parse columnType '%s' for column %s", columnType, columnName));
+        }
+        String fullType = m.group("full");
+        String baseType = m.group("base").trim();
+        if (!Objects.toString(m.group("suffix"), "").isEmpty()) {
+            baseType = String.join(" ", baseType, m.group("suffix").trim());
+        }
+        String[] typeModifiers = {};
+        if (m.group("mod") != null) {
+            typeModifiers = TYPEMOD_PATTERN.split(m.group("mod"));  // TODO: use for decimal/etc precision
+        }
+        boolean isArray = (m.group("array") != null);
+
+        if (baseType.startsWith("_")) {
+            // old-style type specifiers use an _ prefix for arrays
+            // e.g. int4[] would be "_int4"
+            baseType = baseType.substring(1);
+            fullType = fullType.substring(1);
+            isArray = true;
+        }
+
+        if (rawValue.isNull()) {
+            // nulls are null
+            return null;
+        }
+
+        if (isArray) {
+            try {
+                final String dataString = rawValue.asString();
+                PgArray arrayData = new PgArray(connection.get(), connection.get().getTypeInfo().getPGArrayType(fullType), dataString);
+                Object deserializedArray = arrayData.getArray();
+                // TODO: what types are these? Shouldn't they pass through this function again?
+                return Arrays.asList((Object[])deserializedArray);
+            }
+            catch (SQLException e) {
+                LOGGER.warn("Unexpected exception trying to process PgArray ({}) column '{}', {}", columnType, columnName, e);
+            }
+            return null;
+        }
+
+        switch (baseType) {
+            // include all types from https://www.postgresql.org/docs/current/static/datatype.html#DATATYPE-TABLE
+            // plus aliases from the shorter names produced by older wal2json
```

Comment:
Could we group the old names and move them all towards the end? That way we can git rid of them more easily in the future, once we say we don't support them any longer.